src/carma-streets/parse_ros_bags.py

```diff
@@ -37,6 +37,52 @@ def get_detected_objects(ros_bag_file, outputfile, time_offset):
     print(f"Calculated sim time offset: '{time_offset}'")
     
 
+def get_pedestrian_odometry(ros_bag_file, output_file, time_offset):
+    def get_object_with_id(id_, msg):
+        for object_ in msg.objects:
+            if object_.id == id_:
+                return object_
+
+    with rosbag.Bag(ros_bag_file, "r") as bag:
+        messages = [msg for (_, msg, _) in bag.read_messages(topics=["/carla/objects"])]
+
+    output_file.parent.mkdir(exist_ok=True)
+
+    if output_file.exists():
+        print(f"Output file {output_file} already exists. Overwriting file.")
+
+    with open(output_file, "w", newline="") as file:
+        writer = csv.writer(file)
+        writer.writerow(
+            [
+                "cdasim_time_ms",
+                "map_position_x_m",
+                "map_position_y_m",
+                "map_velocity_x_mps",
+                "map_velocity_y_mps",
+            ]
+        )
+
+        for message in messages:
+            pedestrian_odometry = get_object_with_id(221, message)
+
+            if not pedestrian_odometry or math.isclose(
+                pedestrian_odometry.twist.linear.x, 0.0
+            ):
+                continue
+
+            cdasim_time_ms = (message.header.stamp - time_offset).to_sec() * 1_000
+            writer.writerow(
+                [
+                    cdasim_time_ms,
+                    pedestrian_odometry.pose.position.x,
+                    pedestrian_odometry.pose.position.y,
+                    pedestrian_odometry.twist.linear.x,
+                    pedestrian_odometry.twist.linear.y,
+                ]
+            )
+
+
 def main():
```

Comment:
I kind of feel like the file should not inside the carma-streets folder. Maybe  [rosbag_analysis_tools]?(https://github.com/usdot-fhwa-stol/carma-analytics-fotda/tree/develop/src/rosbag_analysis_tools)
We can figure this out in a different PR though so optional